jupyterhub/app.py

```diff
@@ -2240,6 +2240,37 @@ async def init_role_creation(self):
         if self.custom_scopes:
             self.log.info(f"Defining {len(self.custom_scopes)} custom scopes.")
             scopes.define_custom_scopes(self.custom_scopes)
+
+        # remove potentially stale roles from authenticator
+        if self.authenticator.reset_managed_roles_on_startup:
+            deleted = (
+                self.db.query(orm.Role)
+                .filter(orm.Role.managed_by_auth == True)
+                .delete()
+            )
+            if deleted:
+                self.log.info(
+                    f"Deleted {deleted} potentially stale roles previously added by an authenticator"
+                )
+
+        roles_to_load = self.load_roles
+
+        if self.authenticator.manage_roles and self.load_roles:
+            for role_spec in roles_to_load:
+                user_role_assignments = role_spec.get('users', [])
+                group_role_assignments = role_spec.get('groups', [])
+                if user_role_assignments or group_role_assignments:
+                    raise ValueError(
```

Comment:
Make sure to show the offending role and assignment(s) so users know what config to look for

Reply:
Done in 4ec2fe3c191e3d1bd2e6ef915198c2710502801c.